swc-uxp-starter/webpack.config.js

```diff
@@ -0,0 +1,156 @@
+ *  from Adobe.
+ ***************************************************************************/
+
+'use strict';
+
+import { join, resolve } from 'path';
+
+import CopyWebpackPlugin from 'copy-webpack-plugin';
+import MiniCssExtractPlugin from 'mini-css-extract-plugin';
+import { CleanWebpackPlugin } from 'clean-webpack-plugin';
+import { BundleAnalyzerPlugin } from 'webpack-bundle-analyzer';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+const ENV = process.argv.find((arg) => arg.includes('NODE_ENV=production'))
+    ? 'production'
+    : 'development';
+const IS_DEV = ENV === 'development';
+
+const IS_DEV_SERVER = process.argv.find((arg) =>
+    arg.includes('webpack-dev-server')
+);
+const OUTPUT_PATH = IS_DEV_SERVER ? resolve('dist') : resolve('dist');
+
+/**
+ * === Copy static files configuration
+ */
+const copyStatics = {
+    patterns: [
+        {
+            from: 'index.html',
+            context: resolve('./src'),
+            to: OUTPUT_PATH,
+        },
+
+        {
+            from: 'debug.json',
+            context: resolve('./'),
+            to: OUTPUT_PATH,
+        },
+        {
+            from: 'manifest.json',
+            context: resolve('./'),
+            to: OUTPUT_PATH,
+        },
+
+    ],
+};
+
+/**
+ * Plugin configuration
+ */
+const plugins = [
+    new CleanWebpackPlugin(),
+    new CopyWebpackPlugin(copyStatics),
+    new MiniCssExtractPlugin({ filename: '[name].bundle.css' }),
+];
+
+function srcPath(subdir) {
+    return join(__dirname, 'src', subdir);
+}
+
+const shared = (env) => {
+    if (!IS_DEV_SERVER) {
+        plugins.push(
+            new BundleAnalyzerPlugin({
+                analyzerMode: 'static',
+                openAnalyzer: false,
+                reportFilename: '../report/report.html',
+                statsFilename: '../report/stats.json',
+                generateStatsFile: true,
+                statsOptions: {
+                    chunkModules: true,
+                    children: false,
+                    source: false,
+                },
+            })
+        );
+    }
+
+    let cssLoaders = [
+        {
+            loader: 'css-loader',
+            options: { importLoaders: 1 },
+        },
+        {
+            loader: 'postcss-loader',
+            options: {
+                postcssOptions: {
+                    plugins: (loader) => [
+                        require('postcss-import')({
+                            root: loader.resourcePath,
+                        }),
+                        require('postcss-preset-env')({
+                            browsers: 'last 2 versions',
+                        }),
+                        ...(IS_DEV ? [] : [require('cssnano')()]),
+                    ],
+                },
+            },
+        },
+    ];
+
+    return {
+        entry: {
+            main: './src/index.js',
```

Comment:
Developers who are new to webpack might not understand the reference to main.bundle.js. 
Use `index: './src/index.js'` and update index.html accordingly.